tests/cli/test_setup_gate.py

```
@@ -0,0 +1,92 @@
+    `ConfigManager().load()` reads from an empty dir → returns a
+    default `AppConfig(setup_completed=False)`.
+    """
+    monkeypatch.setattr("config.manager.DEFAULT_CONFIG_DIR", tmp_path)
+    return tmp_path
+
+
+@pytest.mark.unit
+@pytest.mark.ci
+@pytest.mark.uses_setup_gate
+@pytest.mark.parametrize("cmd", ["setup", "version", "doctor", "config", "hardware-info"])
+def test_allowlisted_commands_bypass_gate(cmd: str, fresh_config: Path) -> None:
```

Ruff (0.15.12)

[warning] Unused function argument: `fresh_config`

(ARG001)